src/main/kotlin/no/skatteetaten/aurora/cantus/service/DockerHttpClient.kt

```diff
@@ -0,0 +1,293 @@
+package no.skatteetaten.aurora.cantus.service
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.NullNode
+import mu.KotlinLogging
+import no.skatteetaten.aurora.cantus.controller.BadRequestException
+import no.skatteetaten.aurora.cantus.controller.ImageRepoCommand
+import no.skatteetaten.aurora.cantus.controller.SourceSystemException
+import no.skatteetaten.aurora.cantus.controller.blockAndHandleError
+import no.skatteetaten.aurora.cantus.controller.handleError
+import no.skatteetaten.aurora.cantus.controller.handleStatusCodeError
+import no.skatteetaten.aurora.cantus.createObjectMapper
+import org.springframework.http.HttpHeaders.AUTHORIZATION
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import org.springframework.web.reactive.function.client.bodyToMono
+import reactor.core.publisher.Mono
+import java.time.Duration
+
+private val logger = KotlinLogging.logger {}
+
+val manifestV1 = "application/vnd.docker.distribution.manifest.v1+json"
+val manifestV2 = "application/vnd.docker.distribution.manifest.v2+json"
+
+@Service
+class DockerHttpClient(
```

Comment:
Du nevnte dette for meg.
 "make it work, then make it right". Denne trenger en del refaktorering.
Må også legges til tester, mye som ikke dekkes

Reply:
Jupp :)